WordPress/Classes/Services/EditorSettingsService.swift

```diff
@@ -6,10 +6,13 @@ import Foundation
 
 @objc class EditorSettingsService: LocalCoreDataService {
     @objc func syncEditorSettings(for blog: Blog, success: @escaping () -> Void, failure: @escaping (Swift.Error) -> Void) {
-        guard let api = blog.wordPressComRestApi(), let siteID = blog.dotComID?.intValue else {
-            let error = NSError(domain: "EditorSettingsService", code: 0, userInfo: [NSDebugDescriptionErrorKey: "Api or dotCom Site ID not found"])
-            failure(error)
-            return
+        guard let api = blog.wordPressComRestApi() else {
+            // SelfHosted non-jetpack sites won't sync with remote.
+            return success()
+        }
+        guard let siteID = blog.dotComID?.intValue else {
+            let error = NSError(domain: "EditorSettingsService", code: 0, userInfo: [NSDebugDescriptionErrorKey: "dotCom Site ID not found"])
```

Comment:
Not a blocker, but a block with an account should always have an ID, right? We might want to add an assertion here besides the `guard`

Reply:
Right, we already guarded against a `wordPressComRestApi` api, so at this point it should definitely have a dotComID.
I'll add the assertion 👍